### PR TITLE
add data attribute to disable automatic routing

### PR DESCRIPTION
**Pull request by scttdavs**
Hi 👋! So far I love your router! It has all I'm looking for in one for our site. We have a site that will not be 100% SPA, but we would still like to use relative paths everywhere though. Since we have local, staging and production environments, using absolute urls everywhere would be a pain. I think it would be reasonable to allow users to opt out of the automatic History.pushState behavior with relative paths if need be.

### CODE REVIEW

File: src/lib/util/anchor.ts [hunk 1]
```diff
@@ -2,8 +2,6 @@
  * Hook up a click listener to the window that, for all anchor tags
  * that has a relative HREF, uses the history API instead.
  */
-import { stripStart } from "./url";
```
Comment (scttdavs) [inline src/lib/util/anchor.ts]: did not seem to be used in this file anymore.

File: src/lib/util/anchor.ts [hunk 2]
```diff
@@ -22,7 +20,9 @@ export function ensureAnchorHistory () {
 		// We want to allow links to absolute paths without
 		// using the history API. Also, if the target is not this frame we
 		// allow the default behavior (eg. if it should open in a new tab)
-		if (!href.startsWith(location.origin) || ($anchor.target !== "" && $anchor.target !== "_self")) {
+		if (!href.startsWith(location.origin) ||
+		   ($anchor.target !== "" && $anchor.target !== "_self") ||
+		   $anchor.dataset["routerSlot"] === "disable") {
```
Comment (scttdavs) [inline src/lib/util/anchor.ts]: also open to name suggestions for the attribute and value.

File: src/lib/util/anchor.ts [hunk 3]
```diff
@@ -22,7 +20,9 @@ export function ensureAnchorHistory () {
 		// We want to allow links to absolute paths without
 		// using the history API. Also, if the target is not this frame we
 		// allow the default behavior (eg. if it should open in a new tab)
-		if (!href.startsWith(location.origin) || ($anchor.target !== "" && $anchor.target !== "_self")) {
+		if (!href.startsWith(location.origin) ||
```
Comment (scttdavs) [inline src/lib/util/anchor.ts]: @andreasbm according to the comment above, it seems this is also meant to allow links with absolute paths to work as-is and not use History api. However, the href property of the anchor always includes the origin. So this only checks if links are navigating _away_ from the site.

So as is, there is no way to have a link to another part of your site and _not_ use the History api. You would either need a way to opt out of it, like this PR is adding, or do an additional check like:
```js
$anchor.getAttribute("href").startsWith(location.origin) // checks for absolute urls to the current site.
```

File: README.md [hunk 4]
```diff
@@ -176,6 +176,12 @@ Normally an [`anchor element`](https://developer.mozilla.org/en-US/docs/Web/HTML
 
 There are many advantages of using an anchor element, the main one being accessibility.
 
+Alternatively, if you would still like to allow relative links to other parts of your site to navigate as normally, you can opt out of this behavior on a link-by-link basis:
```
Comment (andreasbm) [inline README.md]: I use a readme generator which means the `README.md` file is actually based on the `blueprint.md` file. Would you be able to add the documentation to the `blueprint.md` file instead and run `npm run readme` to generate a new `README.md` file?  I know this hasn't been documented anywhere - I will make sure to do it for future PRs! 👍

File: src/lib/util/anchor.ts (hunk 1 above)
Comment (andreasbm) [reply src/lib/util/anchor.ts]: Good catch 👍

File: src/lib/util/anchor.ts (hunk 2 above)
Comment (andreasbm) [reply src/lib/util/anchor.ts]: I really like the attribute name. I would prefer if we change the value `disable` to `disabled` to make it consistent with the naming of the `disabled` attribute.

File: README.md (hunk 4 above)
Comment (scttdavs) [reply README.md]: done! https://github.com/andreasbm/router-slot/pull/5/commits/d713a3053729607be67744fea2ac5e54ba0bfcba

File: src/lib/util/anchor.ts (hunk 2 above)
Comment (scttdavs) [reply src/lib/util/anchor.ts]: sounds good to me https://github.com/andreasbm/router-slot/pull/5/commits/d713a3053729607be67744fea2ac5e54ba0bfcba

File: blueprint.md [hunk 5]
```diff
@@ -115,6 +115,12 @@ Normally an [`anchor element`](https://developer.mozilla.org/en-US/docs/Web/HTML
 
 There are many advantages of using an anchor element, the main one being accessibility.
 
+Alternatively, if you would still like to allow relative links to other parts of your site to navigate as normally, you can opt out of this behavior on a link-by-link basis:
+
+```html
+<a href="/about" data-router-slot="disable">Go to about!</a>
```
Comment (andreasbm) [inline blueprint.md]: If you also change the value of the attribute to `disabled` here everything looks good 👍

**Pull request merged**